test/cases/~hmr-resilience/error-loader.js

```diff
@@ -0,0 +1,4 @@
+module.exports = function loader() {
```

Comment:
Don't use `~` in filename

Reply:
Any other suggestion then for a moniker to use to indicate tests that are expected to fail?

`Testcases.test.js` works by enumerating directory names and running all the cases, then bailing if one of them has a compilation error.  But here we have a testcase which is expected to have a compilation error. First of its kind.

Reply:
@rjgotten just use `fail` word in test name

Reply:
Adjusted the test pattern in `TestCases.test.js` to `/-fail$/` i.e. any test suffixed with `-fail` is expected to error. And in accordance renamed the testcase to `hmr-resilience-fail`, getting rid of the `~`.